internal/service/alarms/internal/alarms_server.go

```diff
@@ -190,9 +196,149 @@ func (a *AlarmsServer) GetAlarm(ctx context.Context, request api.GetAlarmRequest
+		// Bad request - at least one field is required to patch
+		return api.PatchAlarm400ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			AdditionalAttributes: &map[string]string{
+				"alarmEventRecordId": request.AlarmEventRecordId.String(),
+			},
+			Detail: "at least one field is required to patch",
+			Status: http.StatusBadRequest,
+		}), nil
+	}
+
+	// Check if both fields are included in the request
+	if request.Body.AlarmAcknowledged != nil && request.Body.PerceivedSeverity != nil {
+		// Bad request - only one field is allowed to be patched
+		return api.PatchAlarm400ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			AdditionalAttributes: &map[string]string{
+				"alarmEventRecordId": request.AlarmEventRecordId.String(),
+			},
+			Detail: "either alarmAcknowledged or perceivedSeverity shall be included in a request message content, but not both",
+			Status: http.StatusBadRequest,
+		}), nil
+
+	}
+
+	// Patch perceivedSeverity
+	if request.Body.PerceivedSeverity != nil {
+		perceivedSeverity := *request.Body.PerceivedSeverity
+		// Only the value "5" for "CLEARED" is permitted in a request message content
+		if perceivedSeverity != api.CLEARED {
+			return api.PatchAlarm400ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+				AdditionalAttributes: &map[string]string{
+					"alarmEventRecordId": request.AlarmEventRecordId.String(),
+				},
+				Detail: fmt.Sprintf("only the value %d for CLEARED is permitted in the perceivedSeverity field", api.CLEARED),
+				Status: http.StatusBadRequest,
+			}), nil
+		}
+
+		// Check if the Alarm Event Record has already been cleared
+		if record.PerceivedSeverity == int(perceivedSeverity) {
+			// Nothing to patch
+			return api.PatchAlarm409ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+				AdditionalAttributes: &map[string]string{
+					"alarmEventRecordId": request.AlarmEventRecordId.String(),
+				},
+				Detail: "Alarm record is already cleared",
+				Status: http.StatusConflict,
+			}), nil
+		}
+
+		// Check if associated alarm definition has clearing type "manual". If not, return 409.
+		alarmDefinition, err := a.AlarmsRepository.GetAlarmDefinition(ctx, record.AlarmDefinitionID)
+		if errors.Is(err, utils.ErrNotFound) {
+			return api.PatchAlarm404ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+				AdditionalAttributes: &map[string]string{
+					"alarmEventRecordId": request.AlarmEventRecordId.String(),
+				},
+				Detail: "associated Alarm Definition not found",
+				Status: http.StatusNotFound,
+			}), nil
+		}
+
+		if alarmDefinition.ClearingType != string(apiresources.MANUAL) {
+			return api.PatchAlarm409ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+				AdditionalAttributes: &map[string]string{
+					"alarmEventRecordId": request.AlarmEventRecordId.String(),
+				},
+				Detail: "cannot clear an alarm with clearing type other than MANUAL",
+				Status: http.StatusConflict,
+			}), nil
+		}
```

Comment:
We should check the clearing type as a first step so that we don't proceed further if the type is not "manual".  I would also argue that since we don't currently have any alarms of type "manual" that we probably don't need the other code at all... just this code to always reject because the clearing type is "manual".

Reply:
It seems more natural to me to check the clearing status of the alarm first and then check if it can be cleared. It also saves a DB round trip. But if you have a strong request for the other way. I won't mind changing it.

Regarding the other part, I think for correctness, we should handle clearing of Manual. We don't know what we will have later. Besides, it is just a few lines of code.

Reply:
I don't feel strongly about it, but it seems pointless to validate a request that is not allowed....  also, it gives a client a false impression about what needs to change for their request to succeed.

Reply:
done